lib/nexthop_group.c

```diff
@@ -592,31 +617,60 @@ static bool nexthop_group_parse_nexthop(struct nexthop *nhop,
 	} else
 		nhop->type = NEXTHOP_TYPE_IFINDEX;
 
+	if (labels) {
+		int ret;
+		uint8_t num = 0;
+		mpls_label_t larray[MPLS_MAX_LABELS];
+
+		ret = mpls_str2label(labels, &num, larray);
```

Comment:
Parsing a label string can fail for a number of reasons (e.g. too many labels) and failing silently here can cause confusion. I think we need to provide a detailed error message to the user when something goes wrong (like done on `static_route_leak()`). The "nexthop" command should also fail (i.e. return earlier) when that happens. The reason is that an invalid label string will never become valid under any circumstance (so there's no point in keeping the nexthop around).

Reply:
yes, quite right - I'll update to fix that